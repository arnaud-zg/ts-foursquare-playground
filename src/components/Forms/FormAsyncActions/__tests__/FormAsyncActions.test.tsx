import React from 'react'
import renderer from 'react-test-renderer'
import { FormAsyncActions } from '..'

describe('Component/FormAsyncActions', () => {
  test.each`
    FormAsyncActionsProps
    ${{ hasError: true, onCancel: () => {}, onFailure: () => {}, onRequest: () => {}, onReset: () => {}, onSuccess: () => {} }}
    ${{ hasError: false, onCancel: () => {}, onFailure: () => {}, onRequest: () => {}, onReset: () => {} }}
    ${{ hasError: false, onCancel: () => {}, onFailure: () => {}, onRequest: () => {} }}
    ${{ hasError: false, onCancel: () => {}, onFailure: () => {} }}
    ${{ hasError: false, onCancel: () => {} }}
  `(
    'renders correctly with $FormAsyncActionsProps',
    ({ FormAsyncActionsProps }) => {
      const tree = renderer
        .create(<FormAsyncActions {...FormAsyncActionsProps} />)
        .toJSON()
      expect(tree).toMatchSnapshot()
    }
  )
})
