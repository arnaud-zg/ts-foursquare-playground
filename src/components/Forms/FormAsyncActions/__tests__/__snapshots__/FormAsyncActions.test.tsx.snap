// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component/FormAsyncActions renders correctly with {"hasError": false, "onCancel": [Function onCancel], "onFailure": [Function onFailure], "onRequest": [Function onRequest], "onReset": [Function onReset]} 1`] = `
Array [
  <button
    className="mt-2 bg-white text-gray-800 font-bold rounded border-b-2 shadow-md py-2 px-6 inline-flex items-center border-green-500 hover:border-green-600 hover:bg-green-500 hover:text-white"
    disabled={false}
    onClick={[Function]}
  >
    <span
      className="mr-2"
    >
      Reset
    </span>
  </button>,
  <button
    className="mt-2 ml-2 bg-white text-gray-800 font-bold rounded border-b-2 shadow-md py-2 px-6 inline-flex items-center border-green-500 hover:border-green-600 hover:bg-green-500 hover:text-white"
    disabled={false}
    onClick={[Function]}
    type="submit"
  >
    <span
      className="mr-2"
    >
      Request
    </span>
    <svg
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
        fill="currentcolor"
      />
    </svg>
  </button>,
  <button
    className="mt-2 ml-2 bg-white text-gray-800 font-bold rounded border-b-2 shadow-md py-2 px-6 inline-flex items-center bg-gray-500 cursor-default"
    disabled={true}
  >
    <span
      className="mr-2"
    >
      Success
    </span>
  </button>,
  <button
    className="mt-2 ml-2 bg-white text-gray-800 font-bold rounded border-b-2 shadow-md py-2 px-6 inline-flex items-center border-green-500 hover:border-green-600 hover:bg-green-500 hover:text-white"
    disabled={false}
    onClick={[Function]}
  >
    <span
      className="mr-2"
    >
      Cancel
    </span>
  </button>,
  <button
    className="mt-2 ml-2 bg-white text-gray-800 font-bold rounded border-b-2 shadow-md py-2 px-6 inline-flex items-center border-green-500 hover:border-green-600 hover:bg-green-500 hover:text-white"
    disabled={false}
    onClick={[Function]}
  >
    <span
      className="mr-2"
    >
      Failure
    </span>
  </button>,
]
`;

exports[`Component/FormAsyncActions renders correctly with {"hasError": false, "onCancel": [Function onCancel], "onFailure": [Function onFailure], "onRequest": [Function onRequest]} 1`] = `
Array [
  <button
    className="mt-2 bg-white text-gray-800 font-bold rounded border-b-2 shadow-md py-2 px-6 inline-flex items-center bg-gray-500 cursor-default"
    disabled={true}
  >
    <span
      className="mr-2"
    >
      Reset
    </span>
  </button>,
  <button
    className="mt-2 ml-2 bg-white text-gray-800 font-bold rounded border-b-2 shadow-md py-2 px-6 inline-flex items-center border-green-500 hover:border-green-600 hover:bg-green-500 hover:text-white"
    disabled={false}
    onClick={[Function]}
    type="submit"
  >
    <span
      className="mr-2"
    >
      Request
    </span>
    <svg
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
        fill="currentcolor"
      />
    </svg>
  </button>,
  <button
    className="mt-2 ml-2 bg-white text-gray-800 font-bold rounded border-b-2 shadow-md py-2 px-6 inline-flex items-center bg-gray-500 cursor-default"
    disabled={true}
  >
    <span
      className="mr-2"
    >
      Success
    </span>
  </button>,
  <button
    className="mt-2 ml-2 bg-white text-gray-800 font-bold rounded border-b-2 shadow-md py-2 px-6 inline-flex items-center border-green-500 hover:border-green-600 hover:bg-green-500 hover:text-white"
    disabled={false}
    onClick={[Function]}
  >
    <span
      className="mr-2"
    >
      Cancel
    </span>
  </button>,
  <button
    className="mt-2 ml-2 bg-white text-gray-800 font-bold rounded border-b-2 shadow-md py-2 px-6 inline-flex items-center border-green-500 hover:border-green-600 hover:bg-green-500 hover:text-white"
    disabled={false}
    onClick={[Function]}
  >
    <span
      className="mr-2"
    >
      Failure
    </span>
  </button>,
]
`;

exports[`Component/FormAsyncActions renders correctly with {"hasError": false, "onCancel": [Function onCancel], "onFailure": [Function onFailure]} 1`] = `
Array [
  <button
    className="mt-2 bg-white text-gray-800 font-bold rounded border-b-2 shadow-md py-2 px-6 inline-flex items-center bg-gray-500 cursor-default"
    disabled={true}
  >
    <span
      className="mr-2"
    >
      Reset
    </span>
  </button>,
  <button
    className="mt-2 ml-2 bg-white text-gray-800 font-bold rounded border-b-2 shadow-md py-2 px-6 inline-flex items-center border-green-500 hover:border-green-600 hover:bg-green-500 hover:text-white"
    disabled={false}
    type="submit"
  >
    <span
      className="mr-2"
    >
      Request
    </span>
    <svg
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
        fill="currentcolor"
      />
    </svg>
  </button>,
  <button
    className="mt-2 ml-2 bg-white text-gray-800 font-bold rounded border-b-2 shadow-md py-2 px-6 inline-flex items-center bg-gray-500 cursor-default"
    disabled={true}
  >
    <span
      className="mr-2"
    >
      Success
    </span>
  </button>,
  <button
    className="mt-2 ml-2 bg-white text-gray-800 font-bold rounded border-b-2 shadow-md py-2 px-6 inline-flex items-center border-green-500 hover:border-green-600 hover:bg-green-500 hover:text-white"
    disabled={false}
    onClick={[Function]}
  >
    <span
      className="mr-2"
    >
      Cancel
    </span>
  </button>,
  <button
    className="mt-2 ml-2 bg-white text-gray-800 font-bold rounded border-b-2 shadow-md py-2 px-6 inline-flex items-center border-green-500 hover:border-green-600 hover:bg-green-500 hover:text-white"
    disabled={false}
    onClick={[Function]}
  >
    <span
      className="mr-2"
    >
      Failure
    </span>
  </button>,
]
`;

exports[`Component/FormAsyncActions renders correctly with {"hasError": false, "onCancel": [Function onCancel]} 1`] = `
Array [
  <button
    className="mt-2 bg-white text-gray-800 font-bold rounded border-b-2 shadow-md py-2 px-6 inline-flex items-center bg-gray-500 cursor-default"
    disabled={true}
  >
    <span
      className="mr-2"
    >
      Reset
    </span>
  </button>,
  <button
    className="mt-2 ml-2 bg-white text-gray-800 font-bold rounded border-b-2 shadow-md py-2 px-6 inline-flex items-center border-green-500 hover:border-green-600 hover:bg-green-500 hover:text-white"
    disabled={false}
    type="submit"
  >
    <span
      className="mr-2"
    >
      Request
    </span>
    <svg
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
        fill="currentcolor"
      />
    </svg>
  </button>,
  <button
    className="mt-2 ml-2 bg-white text-gray-800 font-bold rounded border-b-2 shadow-md py-2 px-6 inline-flex items-center bg-gray-500 cursor-default"
    disabled={true}
  >
    <span
      className="mr-2"
    >
      Success
    </span>
  </button>,
  <button
    className="mt-2 ml-2 bg-white text-gray-800 font-bold rounded border-b-2 shadow-md py-2 px-6 inline-flex items-center border-green-500 hover:border-green-600 hover:bg-green-500 hover:text-white"
    disabled={false}
    onClick={[Function]}
  >
    <span
      className="mr-2"
    >
      Cancel
    </span>
  </button>,
  <button
    className="mt-2 ml-2 bg-white text-gray-800 font-bold rounded border-b-2 shadow-md py-2 px-6 inline-flex items-center bg-gray-500 cursor-default"
    disabled={true}
  >
    <span
      className="mr-2"
    >
      Failure
    </span>
  </button>,
]
`;

exports[`Component/FormAsyncActions renders correctly with {"hasError": true, "onCancel": [Function onCancel], "onFailure": [Function onFailure], "onRequest": [Function onRequest], "onReset": [Function onReset], "onSuccess": [Function onSuccess]} 1`] = `
Array [
  <button
    className="mt-2 bg-white text-gray-800 font-bold rounded border-b-2 shadow-md py-2 px-6 inline-flex items-center border-green-500 hover:border-green-600 hover:bg-green-500 hover:text-white"
    disabled={false}
    onClick={[Function]}
  >
    <span
      className="mr-2"
    >
      Reset
    </span>
  </button>,
  <button
    className="mt-2 ml-2 bg-white text-gray-800 font-bold rounded border-b-2 shadow-md py-2 px-6 inline-flex items-center bg-red-500 hover:border-red-600 hover:text-white"
    disabled={false}
    onClick={[Function]}
    type="submit"
  >
    <span
      className="mr-2"
    >
      Request
    </span>
    <svg
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
        fill="currentcolor"
      />
    </svg>
  </button>,
  <button
    className="mt-2 ml-2 bg-white text-gray-800 font-bold rounded border-b-2 shadow-md py-2 px-6 inline-flex items-center border-green-500 hover:border-green-600 hover:bg-green-500 hover:text-white"
    disabled={false}
    onClick={[Function]}
  >
    <span
      className="mr-2"
    >
      Success
    </span>
  </button>,
  <button
    className="mt-2 ml-2 bg-white text-gray-800 font-bold rounded border-b-2 shadow-md py-2 px-6 inline-flex items-center border-green-500 hover:border-green-600 hover:bg-green-500 hover:text-white"
    disabled={false}
    onClick={[Function]}
  >
    <span
      className="mr-2"
    >
      Cancel
    </span>
  </button>,
  <button
    className="mt-2 ml-2 bg-white text-gray-800 font-bold rounded border-b-2 shadow-md py-2 px-6 inline-flex items-center border-green-500 hover:border-green-600 hover:bg-green-500 hover:text-white"
    disabled={false}
    onClick={[Function]}
  >
    <span
      className="mr-2"
    >
      Failure
    </span>
  </button>,
]
`;
